src/global/helpers/native-date-adapter.ts

```diff
@@ -0,0 +1,224 @@
+      this.getYear(d1) === this.getYear(d2)
+    );
+  }
+
+  /** Checks whether the given `obj` is a Date. */
+  public isDateInstance(obj: any): boolean {
+    return obj instanceof Date;
+  }
+
+  /** Checks whether the given `date` is a valid Date. */
+  public isValid(date: Date): boolean {
+    return !!date && !isNaN(date.valueOf());
+  }
+
+  /** Creates a new date by cloning the given one. */
+  public clone(date: Date): Date {
+    return new Date(date.valueOf());
+  }
+
+  /**
+   * Creates a new date adding the number of provided `months` to the provided `date`.
+   * If the calculated month has fewer days than the original one, the date is set to the last day of the month.
+   * E.g. with `date` = new Date(2022, 0, 31) and `months` = 1, it returns new Date(2022, 1, 28).
+   */
+  public addCalendarMonths(date: Date, months: number): Date {
+    const targetMonth = date.getMonth() + months;
+    const dateWithCorrectMonth = new Date(date.getFullYear(), targetMonth, 1, 0, 0, 0, 0);
+    const daysInMonth = this.getNumDaysInMonth(
+      this.getYear(dateWithCorrectMonth),
+      this.getMonth(dateWithCorrectMonth)
+    );
+    // Adapt last day of month for shorter months
+    return new Date(this.clone(date).setMonth(targetMonth, Math.min(daysInMonth, date.getDate())));
+  }
+
+  /** Creates a new date by adding the number of provided `days` to the provided `date`. */
+  public addCalendarDays(date: Date, days: number): Date {
+    const targetDay = date.getDate() + days;
+    return new Date(date.getFullYear(), date.getMonth(), targetDay, 0, 0, 0, 0);
+  }
+
+  /**
+   * Compares two dates.
+   * @param first The first date to compare.
+   * @param second The second date to compare.
+   * @returns 0 if the dates are equal, a number less than 0 if the first date is earlier,
+   *     a number greater than 0 if the first date is later.
+   */
+  public compareDate(first: Date, second: Date): number {
+    return (
+      this.getYear(first) - this.getYear(second) ||
+      this.getMonth(first) - this.getMonth(second) ||
+      this.getDate(first) - this.getDate(second)
+    );
+  }
+
+  /** Creates a Date from a valid input (Date, string or number in seconds). */
+  public deserializeDate(date: Date | string | number): Date | null {
+    if (!date) {
+      return null;
+    }
+    if (typeof date === 'string') {
+      if (!Number.isNaN(+date)) {
+        return new Date(+date * 1000);
+      }
+      return new Date(date);
+    } else if (typeof date === 'number') {
+      return new Date(date * 1000);
+    }
+    if (this.isDateInstance(date) && this.isValid(date)) {
+      return date;
+    }
+    return null;
+  }
+
+  /** Returns the right format for the `valueAsDate` property. */
+  public formatValueAsDate(value: string): Date {
+    const values = value?.split('.');
+    if (!values || values.length <= 2) {
+      return undefined;
+    }
+    return new Date(+values[2], +values[1] - 1, +values[0]);
```

Comment:
should the time not be set to 0 ? otherwise it depends when a user is calling formatValueAsDate(). but maybe I'm wrong

Reply:
It's already 0:00 by default if you don't specify a time, but I can make it explicit